main.tf

```diff
@@ -14,11 +14,28 @@
  * limitations under the License.
  */
 
-terraform {
-  required_version = "~> 0.11.0"
+locals {
+  prefix    = "${var.prefix == "" ? "" : "${var.prefix}-"}"
 }
 
-resource "google_storage_bucket" "main" {
-  project = "${var.project_id}"
-  name    = "${var.bucket_name}"
+resource "google_folder" "folders" {
+  count        = "${length(var.names)}"
+  display_name = "${local.prefix}${element(var.names, count.index)}"
+  parent       = "${var.parent_type}s/${var.parent_id}"
 }
+
+# give project creation access to service accounts
+# https://cloud.google.com/resource-manager/docs/access-control-folders#granting_folder-specific_roles_to_enable_project_creation
+
+resource "google_folder_iam_binding" "owners" {
+  count  = "${var.set_roles ? length(var.names) * length(var.folder_admin_roles) : 0}"
+  folder = "${element(google_folder.folders.*.name, count.index / length(var.folder_admin_roles))}"
+  role   = "${element(var.folder_admin_roles, count.index % length(var.folder_admin_roles))}"
```

Comment:
Should we not use `/` or `%` consistently?

Reply:
Actually this is needed. The '%' operator allows to iterate through the permissions defined in the folder_admin_roles list.